/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

option java_multiple_files = true;
package org.apache.airavata.mft.credential.stubs.s3;

import "common/CredCommon.proto";

message S3Secret {
    string secretId = 1;
    string accessKey = 2;
    string secretKey = 3;
}

message S3SecretGetRequest {
    string secretId = 1;
    org.apache.airavata.mft.credential.stubs.common.AuthToken authzToken = 2;
}

message S3SecretCreateRequest {
    string accessKey = 1;
    string secretKey = 2;
    org.apache.airavata.mft.credential.stubs.common.AuthToken authzToken = 3;
}

message S3SecretUpdateRequest {
    string secretId = 1;
    string accessKey = 2;
    string secretKey = 3;
    org.apache.airavata.mft.credential.stubs.common.AuthToken authzToken = 4;
}

message S3SecretDeleteRequest {
    string secretId = 1;
    org.apache.airavata.mft.credential.stubs.common.AuthToken authzToken = 2;
}